addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1178 @@
+    def action_schedule_meeting(self):
+        """
+        Open meeting's calendar view to schedule meeting on current opportunity.
+        :return dict: dictionary value for created Meeting view
+        """
+        self.ensure_one()
+        res = self.env['ir.actions.act_window'].for_xml_id('calendar', 'action_calendar_event')
+        partner_ids = [self.env.user.partner_id.id]
+        if self.partner_id:
+            partner_ids.append(self.partner_id.id)
+        res['context'] = {
+            'search_default_opportunity_id': self.lead_type == 'opportunity' and self.id or False,
+            'default_opportunity_id': self.lead_type == 'opportunity' and self.id or False,
+            'default_partner_id': self.partner_id.id,
+            'default_partner_ids': partner_ids,
+            'default_team_id': self.team_id.id,
+            'default_name': self.name,
+        }
+        return res
+
+    @api.multi
+    def action_set_lost(self):
+        """ Lost semantic: probability = 0, active = False """
+        return self.write({'probability': 0, 'active': False})
+    # Backward compatibility
+    case_mark_lost = action_set_lost
+
+    @api.multi
+    def action_set_active(self):
+        return self.write({'active': True})
+
+    @api.multi
+    def action_set_unactive(self):
+        return self.write({'active': False})
+
+    @api.multi
+    def action_set_won(self):
+        """ Won semantic: probability = 100 (active untouched) """
+        stages_leads = {}
+        for lead in self:
+            stage_id = lead.stage_find(lead.team_id.id or False, [('probability', '=', 100.0), ('on_change', '=', True)])
+            if stage_id:
+                if stages_leads.get(stage_id):
+                    stages_leads[stage_id].append(lead.id)
+                else:
+                    stages_leads[stage_id] = [lead.id]
+        for stage_id, lead_ids in stages_leads.items():
+            self.browse(lead_ids).write({'stage_id': stage_id})
+        return self.write({'probability': 100})
+    # Backward compatibility
+    case_mark_won = action_set_won
+
+    @api.multi
+    def log_next_activity_1(self):
+        return self.set_next_activity(next_activity_name='activity_1_id')
+
+    @api.multi
+    def log_next_activity_2(self):
+        return self.set_next_activity(next_activity_name='activity_2_id')
+
+    @api.multi
+    def log_next_activity_3(self):
+        return self.set_next_activity(next_activity_name='activity_3_id')
+
+    def set_next_activity(self, next_activity_name):
+        for lead in self:
+            if not lead.last_activity_id:
+                continue
+            next_activity = next_activity_name and getattr(lead.last_activity_id, next_activity_name, False) or False
+            if next_activity:
+                date_action = False
+                if next_activity.days:
+                    date_action = fields.Datetime.to_string(fields.Datetime.now() + timedelta(days=next_activity.days))
```

Comment:
are you sure is it working? 
fields.Datetime.now() is string type. either use python datetime.now or convert fields.Datetime.now into datetime.